Docs/cli (#10)
paastech-cloud/docs: Architecture.md

#### CLI (Command-Line Interface)

> The CLI tool, or Command Line Interface, which is the main way for a User to create and deploy Projects, has been developed in [Go](https://go.dev/). It was chosen for its ease of use, its cross-platform compatibility and its performance. Moreover, the Go language is well suited for CLI development, as it is a compiled language, which allows for a single binary to be distributed, and it is used in many other CLI tools, such as Docker, Kubernetes, Terraform, and more.

Reviewer: ```suggestion
The CLI tool, or Command Line Interface tool, is the main way for a User to create and deploy Projects. It has been developed in [Go](https://go.dev/) and was chosen for its ease of use, its cross-platform compatibility and its performance. Moreover, the Go language is well suited for CLI development, as it is a compiled language, which allows for a single binary to be distributed. It is also used in many other CLI tools, such as Docker, Kubernetes, Terraform, and more.
```

> The CLI tool, or Command Line Interface, which is the main way for a User to create and deploy Projects, has been developed in [Go](https://go.dev/). It was chosen for its ease of use, its cross-platform compatibility and its performance. Moreover, the Go language is well suited for CLI development, as it is a compiled language, which allows for a single binary to be distributed, and it is used in many other CLI tools, such as Docker, Kubernetes, Terraform, and more.

Reviewer: > which allows for a single binary to be distributed
no ? See https://go.dev/doc/install/source#environment GOOS and GOARCH

Reviewer 2: Thanks for the info, this has been removed.

> In order to make this tool in Go, the [Cobra library](https://cobra.dev/) was used. It allows for easy creation of CLI tools, with namely subcommands and flags. It is notably used by the Kubernetes CLI tool, `kubectl`, or even by companies like [Fly.io](https://fly.io/) and [Scaleway](https://www.scaleway.com/) for their own CLI tools.

Reviewer: ```suggestion
In order to make this tool, the [Cobra library](https://cobra.dev/) was used. It allows for easy creation of CLI tools, with namely subcommands and flags. It is notably used by the Kubernetes CLI tool, `kubectl`, or even by companies like [Fly.io](https://fly.io/) and [Scaleway](https://www.scaleway.com/) for their own CLI tools.
```

#### CLI (Command-Line Interface)

> The CLI tool, or Command Line Interface tool, is the main way for a User to create and deploy Projects. IT has been developed in [Go](https://go.dev/) for its ease of use, its cross-platform compatibility and its performance. Moreover, the Go language is well suited for CLI development, as it is a compiled language. It is used in many other CLI tools, such as Docker, Kubernetes, Terraform, and more.

Reviewer 3: ```suggestion
The CLI tool, or Command Line Interface tool, is the main way for a User to create and deploy Projects. It has been developed in [Go](https://go.dev/) for its ease of use, its cross-platform compatibility and its overall great performances. Moreover, the Go language is well suited for CLI development, as it is a compiled language. It is used in many other CLI tools, such as Docker, Kubernetes, Terraform, and more.
```

#### Client sign-up and login process

##### Login process

###### CLI

In order to log in from the CLI tool, the Client must have first created an account on the web frontend since the CLI tool does not allow for account creation.

> This is intentional, and it ensures that creating an account cannot be automated. Using a website can allow for more verifications, like [CAPTCHAs](https://en.wikipedia.org/wiki/CAPTCHA).

Reviewer: ```suggestion
This is intentional, and it ensures that creating an account cannot be automated. Using a website can also allow for more verifications, like [CAPTCHAs](https://en.wikipedia.org/wiki/CAPTCHA).
```

The login process begins with the user typing the `paastech login` command which would prompt them to enter their email address and password. The CLI tool then sends a request to the API to get a JWT token.

> This token is stored locally in the user's config directory and is used for all subsequent commands requiring to be authenticated.

Reviewer 4: Is the line return intentional ? Note that this doesn't go to the next line. Markdown interprets lines 79, 80 and 81 as the same paragraph.

Reviewer 2: Line returns are intentional and help cut the paragraph into small pieces. This will be enforced by Prettier later on.
